Main.hs

```diff
@@ -583,22 +584,34 @@ m4m = (source [I0,I0,I0,I1,I1,I1,I0,I0]) ~> m4a
 
 -- -----------------------------------------------
 
--- understanding unfoldMoore
--- -------------------------
-
--- [copied here for inspection]
-
--- | Construct a Moore machine from a state valuation and transition function
-unfoldMoore1 :: (s -> (b, a -> s)) -> s -> Moore a b
-unfoldMoore1 f = go where
-  go s = case f s of
-    (b, g) -> Moore b (go . g)
-{-# INLINE unfoldMoore1 #-}
-
 {-
+understanding unfoldMoore
+-------------------------
```

Comment:
breaking up the unfoldMoore signature in your explanation — also putting the definition in a comment